server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -2574,6 +2578,475 @@ public void testDeeplyNestedDocument() throws Exception {
+            """));
+
+        assertNotNull(parsedDocument.rootDoc().getField("metrics.service.time"));
+        assertNotNull(parsedDocument.rootDoc().getField("metrics.service.time.min"));
+        assertNotNull(parsedDocument.rootDoc().getField("metrics.service.time.max"));
+        assertNotNull(parsedDocument.rootDoc().getField("metrics.service.time.avg"));
+        assertNull(parsedDocument.dynamicMappingsUpdate());
+    }
+
+    public void testSubobjectsFalseDocsWithInnerObjectMappedAsFieldThatCanParseNativalyObjects() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(mapping(b -> {
+            b.startObject("metrics");
+            {
+                b.field("type", "object").field("subobjects", false);
+                b.startObject("properties");
+                {
+                    b.startObject("service.location");
+                    b.field("type", "geo_point");
+                    b.endObject();
+                }
+                b.endObject();
+            }
+            b.endObject();
+        }));
+        ParsedDocument parsedDocument = mapper.parse(source("""
+            {
+              "metrics": {
+                "service.location" : {
+                    "lat": 41.12,
+                    "lon": -71.34
+                  }
+              }
+            }
+            """));
+        IndexableField location = parsedDocument.rootDoc().getField("metrics.service.location");
+        assertNotNull(location);
+        assertNull(parsedDocument.rootDoc().getField("metrics.service.location.lat"));
+        assertNull(parsedDocument.rootDoc().getField("metrics.service.location.lon"));
+        assertTrue(location instanceof LatLonPoint);
+        Mapper locationMapper = mapper.mappers().getMapper("metrics.service.location");
+        assertNotNull(locationMapper);
+        assertTrue(locationMapper instanceof GeoPointFieldMapper);
+    }
+
+    public void testSubobjectsFalseDocsWithInnerObjectMappedAsNonObject() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(mapping(b -> {
+            b.startObject("metrics");
+            {
+                b.field("type", "object").field("subobjects", false);
+                b.startObject("properties");
+                {
+                    b.startObject("service.time");
+                    b.field("type", "long");
+                    b.endObject();
+                }
+                b.endObject();
+            }
+            b.endObject();
+        }));
+        DocumentParsingException err = expectThrows(DocumentParsingException.class, () -> mapper.parse(source("""
+            {
+              "metrics": {
+                "service.time" : {
+                  "min" : 1
+                }
+              }
+            }
+            """)));
+        String expectedErrorMessage = "[5:5] failed to parse field [metrics.service.time] of type [long] in document with id '1'. "
+            + "Preview of field's value: '{min=1}'";
```

Comment:
Sorry about the confusion: this should not throw an error. it should be possible to map service.time as long, or have a previous document with service time as a long, as well as provide service.time as an object with a subfield. The service.time object should be ignored when it comes to mappings, because subobjects is set to false and only the sub-field should be mapped.